templates/etc/cron/logrotate.erb

```diff
@@ -2,7 +2,12 @@
 # THIS FILE IS AUTOMATICALLY DISTRIBUTED BY PUPPET.  ANY CHANGES WILL BE
 # OVERWRITTEN.
 
+<% if @logrotate_state %>
+OUTPUT=$(<%= @logrotate_path -%> -s <%= @logrotate_state -%> <%= @logrotate_arg -%> 2>&1)
+<% else %>
 OUTPUT=$(<%= @logrotate_path -%> <%= @logrotate_arg -%> 2>&1)
+<% end %>
```

Comment:
Try not to add extra blank lines to the outputted file.  Ideally, users not specifying the new `logrotate_state` parameter wouldn't see puppet make any changes to `/etc/cron/logrotate`.  eg I think you might need to use `<% end -%>` instead of `<% end %>`. (Same with the `if` and `else`).